samples/tools/agentprofiler/README.md

```diff
@@ -0,0 +1,44 @@
+# AutoGen Profiler
+
+AutoGen Profiler is a set of utilities for analyzing and visualizing multi-agent conversations.
```

Comment:
This is a great start towards getting some visibility into what agents are doing. 

Some notes on the readme.

- It would be great to describe in a bit more detail what the current utilities are and what we learn from them. E.g. currently, there seems to be two main utilities - `annotate_chat_history`,  and `draw_profiler_graph`. It would be good to call these out and explain exactly what they do (expected input and output). Also explain the overall mechanism ie .. some 
- It would be good to have additional details on how annotation works. e.g. GPT4 predicts if a predefined set of codes apply to a given message.  
- recommended approach to interpreting the resulting chart with an example.

## Future Ideas 

- support pip install for better developer experience. pip install from the agentprofiler directory will make the package globally available vs updating path.
- support adding agent name to the charts. Implementation wise, this might exploring some chat history logging logic (e.g. register reply function and keep track of sender/receiver name) or utilize [logging](https://github.com/microsoft/autogen/issues/442) capabilities if available.

Reply:
Thank you! Very useful! I will make these changes.

Reply:
Improved readme
Moved profiler to agentchat/contrib to simplify import
other changes are todo

Reply:
added example of group chat being profiled

Reply:
added tests